naga-cli/src/bin/naga.rs

```diff
@@ -465,15 +559,28 @@ fn parse_input(
     input: Vec<u8>,
     params: &Parameters,
 ) -> Result<Parsed, Box<dyn std::error::Error>> {
-    let (module, input_text) = match Path::new(&input_path)
-        .extension()
-        .ok_or(CliError("Input filename has no extension"))?
-        .to_str()
-        .ok_or(CliError("Input filename not valid unicode"))?
-    {
-        "bin" => (bincode::deserialize(&input)?, None),
-        "spv" => naga::front::spv::parse_u8_slice(&input, &params.spv_in).map(|m| (m, None))?,
-        "wgsl" => {
+    let input_kind = match params.input_kind {
+        Some(kind) => kind,
+        None => InputKind::from_str(
+            input_path
+                .extension()
+                .ok_or(CliError::from("Input filename has no extension"))?
```

Comment:
```suggestion
                .context("Input filename has no extension")?
```

With anyhow, this can just be